### PR TITLE
Encode dav path segments for direct preview

### PR DESCRIPTION
**Pull request by PVince81**
Fixes https://github.com/nextcloud/spreed/issues/5672

### CODE REVIEW

File: src/components/MessagesList/MessagesGroup/Message/MessagePart/FilePreview.vue [hunk 1]
```diff
@@ -77,6 +77,9 @@ import Tooltip from '@nextcloud/vue/dist/Directives/Tooltip'
 import Close from 'vue-material-design-icons/Close'
 import PlayCircleOutline from 'vue-material-design-icons/PlayCircleOutline'
 import { getCapabilities } from '@nextcloud/capabilities'
+import {
+	encodePath,
+} from '@nextcloud/paths'
```
Comment (skjnldsv) [inline src/components/MessagesList/MessagesGroup/Message/MessagePart/FilePreview.vue:82]:
```suggestion
import { encodePath } from '@nextcloud/paths'
```

Comment (PVince81) [reply src/components/MessagesList/MessagesGroup/Message/MessagePart/FilePreview.vue:82]: good point

**Comment by PVince81** [comment]
hold on, we have unit tests for this class now, I'll add one to cover the encoding

**Review by ma12-co — APPROVED** [review]

**Comment by PVince81** [comment]
stable21.1: https://github.com/nextcloud/spreed/pull/5674

the stable21 backport will be initiated from that PR

**Pull request merged**